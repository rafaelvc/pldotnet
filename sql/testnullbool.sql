CREATE OR REPLACE FUNCTION returnNullBool() RETURNS boolean AS $$
return null;
$$ LANGUAGE pldotnet;
SELECT returnNullBool();

CREATE OR REPLACE FUNCTION BooleanNullAnd(a boolean, b boolean) RETURNS boolean AS $$
return a&b;
$$ LANGUAGE pldotnet;
SELECT BooleanNullAnd(true, null);
SELECT BooleanNullAnd(null, true);
SELECT BooleanNullAnd(false, null);
SELECT BooleanNullAnd(null, false);
SELECT BooleanNullAnd(null, null);

CREATE OR REPLACE FUNCTION BooleanNullOr(a boolean, b boolean) RETURNS boolean AS $$
return a|b;
$$ LANGUAGE pldotnet;
SELECT BooleanNullOr(true, null);
SELECT BooleanNullOr(null, true);
SELECT BooleanNullOr(false, null);
SELECT BooleanNullOr(null, false);
SELECT BooleanNullOr(null, null);

CREATE OR REPLACE FUNCTION BooleanNullXor(a boolean, b boolean) RETURNS boolean AS $$
return a^b;
$$ LANGUAGE pldotnet;
SELECT BooleanNullXor(true, null);
SELECT BooleanNullXor(null, true);
SELECT BooleanNullXor(false, null);
SELECT BooleanNullXor(null, false);
SELECT BooleanNullXor(null, null);
