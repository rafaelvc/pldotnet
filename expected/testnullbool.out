CREATE OR REPLACE FUNCTION returnNullBool() RETURNS boolean AS $$
return null;
$$ LANGUAGE plcsharp;
SELECT returnNullBool() is NULL;
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION BooleanNullAnd(a boolean, b boolean) RETURNS boolean AS $$
return a&b;
$$ LANGUAGE plcsharp;
SELECT BooleanNullAnd(true, null) is NULL;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullAnd(null, true) is NULL;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullAnd(false, null) is false;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullAnd(null, false) is false;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullAnd(null, null) is NULL;
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION BooleanNullOr(a boolean, b boolean) RETURNS boolean AS $$
return a|b;
$$ LANGUAGE plcsharp;
SELECT BooleanNullOr(true, null) is true;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullOr(null, true) is true;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullOr(false, null) is NULL;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullOr(null, false) is NULL;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullOr(null, null) is NULL;
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION BooleanNullXor(a boolean, b boolean) RETURNS boolean AS $$
return a^b;
$$ LANGUAGE plcsharp;
SELECT BooleanNullXor(true, null) is NULL;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullXor(null, true) is NULL;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullXor(false, null) is NULL;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullXor(null, false) is NULL;
 ?column? 
----------
 t
(1 row)

SELECT BooleanNullXor(null, null) is NULL;
 ?column? 
----------
 t
(1 row)

